packages/@aws-cdk/aws-rds/lib/instance.ts

```diff
@@ -698,6 +728,10 @@ abstract class DatabaseInstanceSource extends DatabaseInstanceNew implements IDa
     this.singleUserRotationApplication = props.engine.singleUserRotationApplication;
     this.multiUserRotationApplication = props.engine.multiUserRotationApplication;
 
+    if (props.domain && !props.engine.engineType.match(/^(mysql|postgres|oracle|sqlserver)/)) {
```

Comment:
Can you actually move this validation into the `IInstanceEngine` implementation? In this case, I believe only `MariaDbInstanceEngine` needs to have this check.

This will require adding a new, optional `domain` property, of type `string`, to `InstanceEngineBindOptions`. The whole thing needs to be similar to how `timezone` is treated there today (up to including the engine type in the error message, like it is for `timezone`).

Reply:
Went ahead and made this change. I did implement the `bindToInstance` method on the `MariaDbInstanceEngine`, calling `super.bindToInstance(...)` to keep the `timezone` property check in one place.